### ISSUE TITLE
Graph Size Normalization from Benchmarking Graph Neural Networks

### ISSUE THREAD

**Issue by RuihongQiu** [rusty1s/pytorch_geometric#1011]
## ❓ Questions & Help

Hi,

I recently read the [Benchmarking Graph Neural Networks](https://arxiv.org/pdf/2003.00982.pdf) paper. There is a normalization called `Graph Size Normalization` that I have never heard of and I can't find any implementation from this repo. So I write my own:

```python
import torch
import torch.nn as nn


class GraphSizeNorm(nn.Module):
    r"""Applies graph size normalization over each individual graph in a batch
    of node features as described in the `"Benchmarking Graph Neural Networks" <https://arxiv.org/abs/2003.00982>`_
    paper

    .. math::
        \mathbf{x}^l_i = \frac{\mathbf{x}^l_i}{\sqrt{|G|}
    """
    def __init__(self):
        super(GraphSizeNorm, self).__init__()

    def forward(self, x, batch=None):
        """"""
        if batch is None:
            batch = torch.zeros(x.size(0), dtype=torch.long, device=x.device)

        sections = torch.bincount(batch)
        x_split = torch.split(x, tuple(sections.cpu().numpy()))
        out = torch.cat([x_i / torch.sqrt(n_node.float()) for n_node, x_i in zip(sections, x_split)], dim=0)
        return out
```

This piece of code can run without error. Could you please have a look and check the code? Thank you so much.

**Comment by rusty1s** [comment]
Yea, that is an interesting approach. Your implementation looks correct to me, although you do not need the for-loop in particular. A more efficient way to implement it would be:
```
def forward(self, x, batch=None):
        """"""
        if batch is None:
            batch = torch.zeros(x.size(0), dtype=torch.long, device=x.device)

        inv_sqrt_deg = torch_geometric.utils.degree(batch, dtype=x.dtype).pow(-0.5)
        return x * inv_sqrt_deg[batch].view(-1, 1)
```
Please let me know if you want to submit a PR by yourself. Otherwise, I would implement it.

Best,
Matthias

**Comment by RuihongQiu** [comment]
Thanks for your reply. It is very great if I could contribute to this project. I will submit a PR very soon.
